apps/ejabberd/src/mod_muc_light_room.erl

```diff
@@ -158,12 +158,20 @@ process_config_set(#config{ raw_config = [{<<"subject">>, _}] } = ConfigReq, Roo
 process_config_set(_ConfigReq, _RoomUS, member, _AffUsers, false) ->
     {error, not_allowed};
 process_config_set(ConfigReq, {_, RoomS} = RoomUS, _UserAff, AffUsers, _AllCanConfigure) ->
-    case mod_muc_light_utils:process_raw_config(
-           ConfigReq#config.raw_config, [], mod_muc_light:config_schema(RoomS)) of
+    Schema = mod_muc_light:config_schema(RoomS),
+    case mod_muc_light_utils:process_raw_config(ConfigReq#config.raw_config, [], Schema) of
         {ok, Config} ->
+            ReadOnlyFields = mod_muc_light:read_only_config_fields(RoomS),
+            ConfigWoReadOnlyFields = mod_muc_light_utils:filter_out_read_only(Config,
```

Comment:
Config update should fail with an error `bad-request` if client tries to change any of read-only fields. Or at least in case of empty change list, because it will trigger room version update and config broadcast... with actually no changes at all.

Reply:
I know that... This is what I had time for now. I'll try to improve it this week.